### PR TITLE
Sequential with slicing capabilities

### LINKED ISSUE #2174: support slicing on nn.Sequential
it'd be great to pick out elements of of a sequential with array slicing:

```
s = nn.Sequential(...)
# Drop the last element
a = s[0:-1]
```

### PR DESCRIPTION
**Pull request by vishwakftw**
This closes https://github.com/pytorch/pytorch/issues/2174

Please let me know what you think about this.

### CODE REVIEW

File: torch/nn/modules/container.py [hunk 1]
```diff
@@ -50,14 +50,21 @@ def __init__(self, *args):
...
-        if not (-len(self) <= idx < len(self)):
-            raise IndexError('index {} is out of range'.format(idx))
-        if idx < 0:
-            idx += len(self)
-        it = iter(self._modules.values())
-        for i in range(idx):
-            next(it)
-        return next(it)
+        if isinstance(idx, slice):
+            ret_seq = Sequential()
```
Comment (gokceneraslan) [inline torch/nn/modules/container.py]: Why not just `return Sequential(*list(self._modules)[idx])`? To preserve named modules?

Comment (vishwakftw) [reply torch/nn/modules/container.py]: Yes. I just assumed that a slice of `Sequential` would require properties of the named modules to be retained.

**Comment by gokceneraslan** [comment]
Why is this closed?

**Comment by vishwakftw** [comment]
The branch was inactive. The branch got deleted accidentally too. I will open a PR if you want me to.

**Pull request closed without merging**